app/models/repository.rb

```diff
@@ -29,12 +29,17 @@
 
 class Repository < ActiveRecord::Base
   include Redmine::Ciphering
+  include OpenProject::Scm::ManageableRepository
 
   belongs_to :project
   has_many :changesets, order: "#{Changeset.table_name}.committed_on DESC, #{Changeset.table_name}.id DESC"
 
   before_save :sanitize_urls
 
+  # Managed repository lifetime
+  after_save :create_managed_repository, if: Proc.new { |repo| repo.managed? }
+  after_destroy :delete_managed_repository, if: Proc.new { |repo| repo.managed? }
```

Comment:
I actually had that using notifications that called the repository services for creation and deletion, and I liked that separation, however that comes at one major price: Using a hook in the repository gives us a free transaction, which will safely abort the repository saving/deletion if an error occurred in here.

Do you see a way to provide this with notifications without manually searching and heuristically 'reverting' the change ?